gammapy/data/observations.py

```diff
@@ -635,6 +635,27 @@ def select_time(self, time_intervals):
     def _ipython_key_completions_(self):
         return self.ids
 
+    def split(self, groups_id):
```

Comment:
Maybe rename to `.group_by_id()` or similar? I think it would be more clear and similar to e.g. `Table.group_by()`...

Reply:
This is a bit of an arbitrary idea: put one could also mirror the `Table` API. So `Observations` will get an optional `.groups` argument, such that the  following works:

```python3
observations = obersvations.group_by_id(groups_id)

for group in observations.groups:
    for obs in group:
        ...

```

Reply:
- renamed `group_by_label` to avoid confusion with the `obs_id`
- maybe we can keep the all the groups functionalities for later because it seems more complex than just returning a list of table.